R/expand.R

```diff
@@ -170,7 +170,12 @@ expand_grid <- function(..., .name_repair = "check_unique") {
     each <- n / cumprod(ns)
     times <- n / each / ns
 
-    out <- pmap(list(x = dots, each = each, times = times), vec_repeat)
+    out <- pmap(
+      list(x = dots, each = each, times = times),
+      function(x, each, times) {
+        vec_rep(vec_rep_each(x, each), times)
+      }
+    )
```

Comment:
Might be slightly cleaner as two sets of `map2()` calls, like in https://github.com/r-lib/vctrs/issues/1325#issuecomment-786148657

```r
times <- n / cumprod(ns)
out <- map2(dots, times, vec_rep_each)
times <- n / times / ns
out <- map2(out, times, vec_rep)
```